02/studenttests.ml

```diff
@@ -256,13 +261,25 @@ let cc_sub = fun (dest:Int64.t) (src:Int64.t) -> Gradedtests.test_machine
   ;InsB0 (Subq, [Imm (Lit src); ~%Rax]);InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag
   ]
 
+let cc_imul = fun (a:int64) (b:int64) -> Gradedtests.test_machine
+  [InsB0 (Movq, [Imm (Lit a); ~%Rax]);InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag
+  ;InsB0 (Imulq, [Imm (Lit b); ~%Rax]);InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag
+  ]
+
+let cc_neg = fun (a:int64) -> Gradedtests.test_machine
+  [InsB0 (Movq, [Imm (Lit a); ~%Rax]);InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag
+  ;InsB0 (Negq, [~%Rax]);InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag;InsFrag
+  ]
+
 let condition_flag_set_tests = 
   [ (* Logic instruction. *)
     ("cc_and", Gradedtests.cs_test 2 (cc_and ()) (false, false, false))
   ; ("cc_or",  Gradedtests.cs_test 2 (cc_or  ()) (false, false, false))
   ; ("cc_xor", Gradedtests.cs_test 2 (cc_xor ()) (false, false, false))
   ; ("cc_not", Gradedtests.csi_test 2 (cc_not ()))
-  ; ("cc_not_negative", Gradedtests.cs_test 3 (cc_not_negative ()) (false, false, false))
+  ; ("cs_not", Gradedtests.csi_test 2 (cc_not ()))
+  ; ("cc_and_arg =0", Gradedtests.cs_test 2 (cc_and_arg 0 12345) (false, false, true))
+  ; ("cc_and_arg =-1", Gradedtests.cs_test 2 (cc_and_arg (-1) (-1)) (false, true, false))
```

Comment:
Can you use snake case and avoid spaces? Just for consistency
You may use eq for = and neg for -

Reply:
I've changed it, but I personally do not find these names to be very helpful. For example, seeing that "cc_sar_0_a: OF:false SF:false ZF:false"  failed (randomly picked) does not tell me what's really going on without reading the test code anyway. In my experience, BDD leads to more understandable test descriptions, but that might be overkill for these projects.